privacyidea/api/lib/postpolicy.py

```diff
@@ -761,3 +761,13 @@ def is_authorized(request, response):
             raise ValidateError("User is not authorized to authenticate under these conditions.")
 
     return response
+
+
+def token_serial_from_response_g(request, response):
+    """
+    This is decorator function fetches the token serial from the response
+    and adds it to the flask g object. It decorates api functions where the serial
+    is only determined during the call, e.g. validate/check.
+    """
+    if g.serial is not None:
+        g.serial = response.json.get("detail", None).get("serial")
```

Comment:
This will raise an exception, if the response does not contain a "detail", since it results in ``None.get("serial")``.

You should do 

    response.json.get("detail", {}).get("serial")

Reply:
I already experienced this. Changed.